### PR TITLE
Need to install codecov

### PR DESCRIPTION
**Pull request by dgasmith**
Missing codecov dependency currently.

### CODE REVIEW

**Comment by Lnaden** [comment]
I actually have `codecov` installed in the main `.travis.yml` file since `codecov` does not appear on the default anaconda channel. I wanted to keep the place where its installed in one file rather than have it split. Its probably worth adding a comment in the `setup.sh` file there though explaining that `codecov` is installed later for that exact reason.

https://github.com/choderalab/cookiecutter-compchem/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/.travis.yml#L40-L44

**Comment by dgasmith** [comment]
Ah, I see. I went through the pip only version which will not install codecov.

**Comment by Lnaden** [comment]
Oh! You're right! I did miss it on that path! I think I had it there at one point but it went missing somewhere. 

> Might be good to only install codecov through pip?

That would simplify the script a bit, but since the option is to prefer packages from `conda-forge` and use `pip` only as fallback, I tried to keep with that as much as possible.

**Comment by Lnaden** [comment]
I did get it on the `appveyor.yml` file though for all 3 paths

**Comment by Lnaden** [comment]
You should be good to merge PRs on this repo as well now, so if this is the only thing for this PR, go ahead and try to merge. It will be a good test

**Comment by dgasmith** [comment]
I took a deep look at this and have made the following changes:
 - Simplified the logic trees a bit
 - Attempted to make it clear that the `travis-ci/before_install.py` is for setup purposes only
 - Removes Py2.7 testing
 - Always install codecov via pip to make things quite a bit easier and allow a pip option

File: {{cookiecutter.repo_name}}/.travis.yml [hunk 1]
```diff
@@ -1,59 +1,58 @@
...
     # Build and install package
   - conda build --python=$PYTHON_VER devtools/conda-recipe
   - conda install --yes --use-local {{cookiecutter.repo_name}}
-{% elif cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
-  - pip install -e .
 {% endif %}
 
 script:
-  - pytest -v --cov={{cookiecutter.repo_name}} --cov-config setup.cfg {{cookiecutter.repo_name}}/tests/
+  - pytest -v --cov={{cookiecutter.repo_name}} {{cookiecutter.repo_name}}/tests/
```
Comment (Lnaden) [inline {{cookiecutter.repo_name}}/.travis.yml]: Do you not need the `--cov-config` if a file called `setup.cfg` is present and has the `[coverage:run]` header?

Comment (dgasmith) [reply {{cookiecutter.repo_name}}/.travis.yml]: Yea, `coverage` can correctly find the data in `setup.cfg` without it.

**Comment by dgasmith** [comment]
Good to merge when ready.

**Pull request merged**